clouddriver-kubernetes/src/main/java/com/netflix/spinnaker/clouddriver/kubernetes/caching/agent/KubernetesCoreCachingAgent.java

```diff
@@ -59,4 +59,20 @@ public KubernetesCoreCachingAgent(
   protected List<KubernetesKind> primaryKinds() {
     return credentials.getGlobalKinds();
   }
+
+  /**
+   * See the comment on the super method for more details about why this function exists.
+   *
+   * <p>As noted there, we want the KubernetesCoreCachingAgent to handle all requests for pending
+   * on-demand cache refreshes to avoid duplicate work. But as users can have multiple cache threads
+   * for a given account, there may be multiple KubernetesCoreCachingAgent's for a single account,
+   * all of which would duplicate work if we sent the work to all of them.
+   *
+   * <p>In order to minimize duplicate work, we'll elect the agent with index 0 to handle all of
+   * these requests, and have it return all pending on-demand refresh requests (not just ones
+   * related to its slice of namespaces).
+   */
+  protected boolean handlePendingOnDemandRequests() {
```

Comment:
I'm late and probably unaware of some coding convention or other detail...but doesn't `@Override` belong here?

Reply:
Fair enough, it probably should be there for documentation purposes (though of course wouldn't affect the run-time behavior).  Given that this is merged, not going to open another PR just for that but next time I'm in there will try to remember to update it.

Reply:
Sounds good.  Thanks.